include/emp/prefab/ControlPanel.hpp

```diff
@@ -0,0 +1,331 @@
+#ifndef EMP_CONTROL_PANEL_HPP
+#define EMP_CONTROL_PANEL_HPP
+
+#include "emp/base/optional.hpp"
+#include "emp/base/vector.hpp"
+
+#include "emp/prefab/ButtonGroup.hpp"
+#include "emp/prefab/FontAwesomeIcon.hpp"
+#include "emp/prefab/ToggleButtonGroup.hpp"
+
+#include "emp/tools/string_utils.hpp"
+
+#include "emp/web/Animate.hpp"
+#include "emp/web/Element.hpp"
+#include "emp/web/Div.hpp"
+
+namespace emp::prefab {
+
+  namespace internal {
+    using checker_func_t = std::function<bool(const web::Animate &)>;
+    /**
+     * Shared pointer held by instances of ControlPanel class representing
+     * the same conceptual ControlPanel DOM object.
+     * Contains state that should persist while ControlPanel DOM object
+     * persists.
+     */
+    class ControlPanelInfo : public web::internal::DivInfo {
+
+      // The unit for rate of refresh
+      std::string refresh_unit;
+
+      // A list of refresh rates with current value associated with current unit
+      std::map<std::string, int> refresh_rates{
+        {"MILLISECONDS", 100}, {"FRAMES", 5}
+      };
+
+      // Map units to cool refresh checkers
+      const std::map<std::string, const checker_func_t> refresh_checkers{
```

Comment:
hmm... it looks like you're only allowing one refresh checker at a time?

What I would do is make two functor classes as public members of this class

ControlPanel::MillisecondRefresher

ControlPanel::FrameRefresher

Both functor classes would take a value as their constructor

instead of storing refresh_rates, refrhesh_units, and refresh_checkers

I would just store a std::function refresh_checker